static/js/chart/draw_scatter.ts

```diff
@@ -358,6 +361,31 @@ function addDensityLegend(
   );
 }
 
+/**
+ * Gets the most recent population value for the given point.
+ * Returns undefined if no population is set
+ * @param point scatter plot Point object
+ */
+function getPointPopulation(point: Point): number | undefined {
+  const xPopDate = point.xPopDate || 0;
+  const yPopDate = point.yPopDate || 0;
+  return xPopDate > yPopDate ? point.xPop : point.yPop;
+}
+
+/**
+ * Calculates scatter plot point size
+ * @param point scatter plot Point object
+ * @param pointSizeScale d3 scale for sizing point based on population
+ */
+function calculatePointSize(
+  point: Point,
+  pointSizeScale: ScatterScale
+): number {
+  const population = getPointPopulation(point);
+  const pointSize = pointSizeScale(population);
```

Comment:
since population could potentially return undefined, curious what `pointSizeScale(undefined)` will return?

Reply:
Good callout- i added a comment to the function to clarify this.

If the population is undefined, the result from the scale will also be undefined. That is accounted for in the next line:

```
return pointSize || DEFAULT_POINT_SIZE;
```

Example
```
$ node
> d3 = require('d3')
> scale = d3.scaleLinear().domain([5,10]).range([10,20])
> scale(5)
10
> scale(undefined)
undefined
```